pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -135,6 +138,12 @@ private QueryContext(@Nullable String tableName, @Nullable QueryContext subquery
     _limit = limit;
     _offset = offset;
     _queryOptions = queryOptions;
+    if (_queryOptions != null) {
```

Comment:
Suggest not changing it in this PR for the following reasons:
1. These options only apply to server but not broker and we don't want to pay the overhead on the broker
2. We may add some query options on the fly which won't be applied here
3. Parsing all the options in the same place is easier to track

Since `nullHandlingEnabled` is needed on both broker and server side, we may call the setter in 2 places